### PR TITLE
Change is to == to satisfy python 3.8 SyntaxWarning

### PR DESCRIPTION
**Pull request by mneil**
Python 3.8 raises a syntax warning when using is to compare
integers or string literals.

Fixes https://github.com/jmespath/jmespath.py/issues/187

### CODE REVIEW

**Comment by mneil** [comment]
Doh! I figured the fix wasn't that simple. If `is` is truly what's needed here, maybe the warnings should be suppressed for python 3.8?

**Comment by andersk** [comment]
`type(x) is int and (x == 0 or x == 1)` should work.

**Comment by mneil** [comment]
Python 2.6 and 3.3 are failing to Download.

**Comment by mneil** [comment]
I don't know why Travis isn't showing this status here but the tests for this pass on all versions with the exception of 2.6 and 3.3 because Travis is failing to DL those versions.

**Comment by mhoonjeon** [comment]
Any plan to merge this and release new version soon? My logs are cluttered with this warnings. I know it could just ignore this but new release can be a better solution :)

**Comment by gene1wood** [comment]
@jamesls Any chance you or someone else at AWS could look at this. This fixes a bothersome bug now that we live in a Python 3.8 world.

File: jmespath/visitor.py [hunk 1]
```diff
@@ -29,9 +29,9 @@ def _is_special_integer_case(x, y):
     # Also need to consider that:
     # >>> 0 in [True, False]
     # True
-    if x is 0 or x is 1:
+    if type(x) is int and (x == 0 or x == 1):
```
Comment (alanyee) [inline jmespath/visitor.py]:
```suggestion
    if isinstance(x, int) and (x == 0 or x == 1):
```
PEP 8

File: jmespath/visitor.py [hunk 2]
```diff
@@ -29,9 +29,9 @@ def _is_special_integer_case(x, y):
...
         return y is True or y is False
-    elif y is 0 or y is 1:
+    elif type(y) is int and (y == 0 or y == 1):
```
Comment (alanyee) [inline jmespath/visitor.py]:
```suggestion
    elif isinstance(y, int) and (y == 0 or y == 1):
```

File: jmespath/visitor.py [hunk 3]
```diff
@@ -257,7 +257,7 @@ def visit_and_expression(self, node, value):
 
     def visit_not_expression(self, node, value):
         original_result = self.visit(node['children'][0], value)
-        if original_result is 0:
+        if type(original_result) is int and original_result == 0:
```
Comment (alanyee) [inline jmespath/visitor.py]:
```suggestion
        if isinstance(original_result, int) and original_result == 0:
```

File: jmespath/visitor.py (hunk 1 above)
Comment (andersk) [reply jmespath/visitor.py]: No, as you can see from the comment preceding the code, or by running the tests, we actually *require* the behavior that PEP 8 is trying to guard against.

```console
>>> type(True) is int
False
>>> isinstance(True, int)
True
```

(This is at least a defined part of the language standard where the behavior of `is 0` and `is 1` is not.)

File: jmespath/visitor.py (hunk 2 above)
Comment (andersk) [reply jmespath/visitor.py]: No, see above.

File: jmespath/visitor.py (hunk 3 above)
Comment (andersk) [reply jmespath/visitor.py]: No, see above.

**Comment by mneil** [comment]
Is there a contribution guide anywhere? And can you point me to the failing tests? I'm only seeing coverage reports. Sorry if I'm missing it. Locally I'm running nose against the changes and they pass - however I'm only testing 3.8 which may explain why mine pass and you see them failing.

**Comment by gene1wood** [comment]
The reason that the [Python 2.6 build](https://travis-ci.org/jmespath/jmespath.py/jobs/605567537?utm_medium=notification&utm_source=github_status) failed and the [Python 3.3 build](https://travis-ci.org/jmespath/jmespath.py/jobs/605567539?utm_medium=notification&utm_source=github_status) failed is that [Travis began gradually changing the default build for Linux from Ubuntu 14.04 to Ubuntu 16.04](https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment) and they don't support Python 2.6 or Python 3.3 under their Ubuntu 16.04 image (see travis-ci/travis-ci#9133). Once #166 is merged this problem goes away as support for Python 2.6 and 3.3 is dropped and [`.travis.yml` is fixed](https://github.com/jmespath/jmespath.py/pull/166/files#diff-354f30a63fb0907d4ad57269548329e3)

The reason that the [Python 3.8 build](https://travis-ci.org/jmespath/jmespath.py/jobs/605567543?utm_medium=notification&utm_source=github_status) fails is because of a bug in [Coverage.py](https://github.com/nedbat/coveragepy) captured in nedbat/coveragepy#772

**Comment by AlJohri** [comment]
Can we please merge this ASAP? The awscli warnings are mighty annoying now that homebrew has upgraded awscli to use python@3.8.

**Comment by stephenrjohnson** [comment]
Any way to suppress these warning with an env variable for the time being ?

**Comment by opub** [comment]
> Any way to suppress these warning with an env variable for the time being ?

export PYTHONWARNINGS=ignore

**Comment by jamesls** [comment]
Hey everyone, sorry for the delay, I'm looking at this now.  Seems like there's a quite a few comments so I'm catching up on this.

**Comment by jamesls** [comment]
Ok the patch itself looks good so I'm going to go ahead and merge this.  I manually verified that 2.6 still passes all the tests, but I do want to get travis back up and working before cutting a release, and I don't want to drop py26/py33 as part of this release.  I'd like to just fix the immediate need of supporting 3.8, plus we haven't done a deprecation period yet (#175).  I'll follow up on that PR as well after this.  I did manually verify that that py26 still passes all the tests.

```
(venv26) ~/Source/jmespath (stelligent-develop) $ python --version
Python 2.6.9
(venv26) ~/Source/jmespath (stelligent-develop) $ nosetests tests/
........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 968 tests in 1.022s

OK
```

So to summarize:

* Get this PR merged.
* Fix .travis.yml to work on 26/33.  From the linked issue we might be able to lock to `trusty` to get the builds passing.
* Cut new release with 3.8 support.  This should fix the immediate need of 3.8 support.  Given this isn't adding new functionality or dropping anything (it's almost a "bug fix" to support 3.8 without generating any warnings) I'll call this `0.9.5`.

Thanks @mneil for the PR,  @gene1wood for the deep dive, and everyone else for chiming in on this PR!
